File: public/dsa/searching/is_perfect_square/naive/is_perfect_square.py
class Solution:
    def isPerfectSquare(self, n: int) -> bool:
        for v in range(1, n + 1):
            sq = v * v
            if sq == n:
                return True
            elif sq > n:
                return False


# tests-start
sol = Solution()
print(sol.isPerfectSquare(25))

sol = Solution()
print(sol.isPerfectSquare(30))
# tests-end

'''output
True
False
'''

File: public/dsa/searching/is_perfect_square/naive/is_perfect_square.cpp
#include <bits/stdc++.h>
using namespace std;

bool isPerfectSquare(int num) {
    for (int v = 1; v <= num; v++) {
        int sq = v * v;
        if (sq == num) {
            return true;
        } else if (sq > num) {
            return false;
        }
    }
}

// tests-start
int main() {
    cout << (isPerfectSquare(25) ? "true" : "false") << endl;
    cout << (isPerfectSquare(30) ? "true" : "false") << endl;
    return 0;
}

// tests-end

/*output
true
false
*/